_posts/2019-04-30-accessing-kafka-part-3.md

```diff
@@ -0,0 +1,165 @@
+From this broker, they will get the metadata which will contain the DNS names of the per-broker routes.
+The Kafka clients will use these addresses to connect to the routes dedicated to the specific broker.
+And the router will again route it through the corresponding service to the right pod.
+
+As explained in the previous section, the routers main use-case is routing of HTTP(S) traffic.
+Therefore it is always listening on the ports 80 and 443.
+Since Strimzi is using the TLS passthrough functionality, it means that:
+* The port will be always 443 as the port used for HTTPS.
+* The traffic will **always use TLS encryption**.
+
+Getting the address where to connect with your client is easy.
+As mentioned above, the port will be always 443.
+This is often a cause of issues, when users try to connect to port 9094 instead of 443.
+But 443 is always the correct port number with OpenShift Routes.
+And you can find the host in the status of the `Route` resource (replace `my-cluster` with the name of your cluster):
+
+```
+oc get routes my-cluster-kafka-bootstrap -o=jsonpath='{.status.ingress[0].host}{"\n"}'
+```
+
+By default, the DNS name of the route will be based on the name of the service it points to and on the name of the OpenShift project.
+So for example for my Kafka cluster named `my-cluster` running in project named `myproject`, the default DNS name will be something like `my-cluster-kafka-bootstrap-myproject.mydomain.io`.
+
+Since it will always use TLS, you will always have to configure the TLS also in your Kafka clients.
+This includes getting the TLS certificate from the broker and configuring it in the client.
+You can use following commands to get the CA certificate used by the Kafka brokers and import it into Java keystore file which can be used with Java applications (replace `my-cluster` with the name of your cluster):
+
+```
+oc extract secret/my-cluster-cluster-ca-cert --keys=ca.crt --to=- > ca.crt
+keytool -import -trustcacerts -alias root -file ca.crt -keystore truststore.jks -storepass password -noprompt
+```
+
+With the certificate and address, you can connect to the Kafka cluster.
+The following example uses the `kafka-console-producer.sh` utility which is part of Apache Kafka:
+
+```
+bin/kafka-console-producer.sh --broker-list <route-address>:443 --producer-property security.protocol=SSL --producer-property ssl.truststore.password=password --producer-property ssl.truststore.location=./truststore.jks --topic <your-topic>
+```
+
+For more details, see the [Strimzi documentation](https://strimzi.io/docs/latest/full.html#proc-accessing-kafka-using-routes-deployment-configuration-kafka).
+
+# Customizations
+
+As explained in the previous section, the routes get by default automatically assigned DNS names based on the name of your cluster and namespace.
+But you can customize this and specify your own DNS names:
```

Comment:
It's best to avoid sentences starting with "But...". In this case just include the but in the previous sentence.